### PR TITLE
Downgrade the-battle-for-wesnoth from 1.17.0 to 1.16.2

### PR DESCRIPTION
**Pull request by gggto**
The Battle for Wesnoth minor version number is by convention even for stable releases and odd for dev releases.

This regex ensures only even releases are captured by the livecheck.

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Comment by bevanjkay** [comment]
Thanks @gggto - I'll defer to @samford for the best way to proceed.

File: Casks/the-battle-for-wesnoth.rb [hunk 1]
```diff
@@ -10,7 +13,7 @@
 
   livecheck do
     url :homepage
-    regex(/href=.*?Wesnoth[._-]v?(\d+(?:\.\d+)+[a-z]?)\.dmg/i)
+    regex(/href=.*?Wesnoth[._-]v?(\d+(?:\.\d*[02468])+(?:\.\d+)*[a-z]?)\.dmg/i)
```
Comment (samford) [inline Casks/the-battle-for-wesnoth.rb:16]:
```suggestion
    regex(/href=.*?Wesnoth[._-]v?(\d+\.\d*[02468](?:\.\d+)*[a-z]?)\.dmg/i)
```

The part that matches the even-numbered minor version doesn't need to repeat, as the following `(?:\.\d+)*` is intended to match any numeric parts after the minor. If we address that issue, this regex will be in line with existing regexes where we match only versions with an even-numbered minor.

**Comment by samford** [comment]
The `livecheck` block should be good now but the latest stable version is 1.16.2, so we probably want to downgrade this, too.

Unless something has changed recently, we only use `# typed: false` and `# frozen_string_literal: true` comments in Homebrew/brew files, not formulae or casks. The only cask with these comments is `dmenu-mac` (added in #108028) and no formulae have these annotations. We should remove these comments here (and `dmenu-mac` in a separate PR) unless/until this is something that we decide to roll out to all first-party formulae/casks.

File: Casks/the-battle-for-wesnoth.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# typed: false
+# frozen_string_literal: true
+
```
Comment (samford) [inline Casks/the-battle-for-wesnoth.rb:3]:
```suggestion
```

**Comment by samford** [comment]
@bevanjkay I think this should be in good shape now but I'll leave final review up to you (or other cask maintainers).

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
`brew` was probably parsing the file as `brew` code and not a cask. To avoid that, do
```
brew audit --cask the-battle-for-wesnoth
```
Importantly, make sure to avoid passing a path to the cask file to `brew audit`, or passing a name that ends in `.rb`.

**Pull request merged**